lib/maths/analytics/CBoostedTreeHyperparameters.cc

```diff
@@ -468,10 +492,13 @@ bool CBoostedTreeHyperparameters::captureBest(const TMeanVarAccumulator& testLos
     // We capture the parameters with the lowest error at one standard
     // deviation above the mean. If the mean error improvement is marginal
     // we prefer the solution with the least variation across the folds.
-    double testLoss{lossAtNSigma(1.0, testLossMoments) + this->modelSizePenalty(numberNodes)};
+    double testLoss{lossAtNSigma(1.0, testLossMoments)};
+    double penalizedTestLoss{testLoss + this->modelSizePenalty(numberNodes)};
+    double bestPenalizedTestLoss{m_BestForestTestLoss + this->modelSizePenalty(numberNodes)};
```

Comment:
```suggestion
    double bestPenalizedTestLoss{m_BestForestTestLoss + this->modelSizePenalty(m_BestForestNumberNodes)};
```
Shouldn't this be `m_BestForestNumberNodes` now?